cms/middleware/toolbar.py

```diff
@@ -27,7 +27,11 @@ def is_cms_request(self, request):
 
         if internal_ips:
             client_ip = get_request_ip(request)
-            if client_ip not in internal_ips:
+            try:
+                if client_ip not in internal_ips:
+                    return False
+            except TypeError:
```

Comment:
Please use `forms.GenericIPAddressField` to clean/validate the ip value.
Something like:

```python
try:
    client_ip = forms.GenericIPAddressField().clean(client_ip)
except ValidationError:
    return False
else:
    return client_ip in internal_ips
```